### DISCUSSION TITLE
Using populate with pagination

### DISCUSSION THREAD

**Discussion by buzzcore** [hagopj13/node-express-boilerplate#97] [Q&A] [answered]
As a beginner in Node.js I am loving this boilerplate. However I want to use mongo's .populate method to populate a relationship. How do I go about this whilst still making use of the paginate plugin?

**Comment by pkester** [comment, marked as answer]
Hello, I am also new to this project and the entire ecosystem but here is how I did it based at the controller level. Not sure about best practice, maybe someone else with more experience can chime in.

```
const getProjects = catchAsync(async (req, res) => {
  const filter = pick(req.query, ['lastname', 'city']);
  const options = pick(req.query, ['sortBy', 'limit', 'page']);
  options.populate = 'user'; 
  const result = await projectService.queryProjects(filter, options);
  res.send(result);
});
```
